src/main/java/study/calculator/Calculator.java

```diff
@@ -1,43 +1,40 @@
 package study.calculator;
 
+import java.util.Objects;
+
 public class Calculator {
     private static final String DELIMITER = " ";
-    private Operator operator;
-
-    Calculator() {
-        operator = new Operator();
-    }
 
-    public Integer calculate(String expression) {
+    public float calculate(String expression) {
         validate(expression);
 
         String[] parts = expression.trim().split(DELIMITER);
-        if(parts.length < 3) {
+        if (parts.length < 3) {
             throw new IllegalArgumentException();
         }
-        Integer result = Integer.parseInt(parts[0]);
+        float result = Float.parseFloat(parts[0]);
         for (int i = 2; i < parts.length; i = i + 2) {
-            result = operate(parts[i-1], result, Integer.parseInt(parts[i]));
+            result = operate(parts[i - 1], result, Float.parseFloat(parts[i]));
         }
 
         return result;
     }
 
     private void validate(String expression) {
-        if(expression == null) {
+        if (Objects.isNull(expression)) {
             throw new IllegalArgumentException();
         }
-        if(expression.trim().isEmpty()) {
+        if (expression.trim().isEmpty()) {
             throw new IllegalArgumentException();
         }
     }
 
-    private Integer operate(String stringOperator, Integer leftSide, Integer rightSide) {
-        if ("+".equals(stringOperator)) return operator.plus(leftSide, rightSide);
-        else if ("-".equals(stringOperator)) return operator.minus(leftSide, rightSide);
-        else if ("/".equals(stringOperator)) return operator.divide(leftSide, rightSide);
-        else if ("*".equals(stringOperator)) return operator.multiply(leftSide, rightSide);
-
-        throw new IllegalArgumentException();
+    private float operate(String stringOperator, float leftSide, float rightSide) {
+        Operand operand = new Operand(leftSide, rightSide);
+        Operator operator = Operator.symbolOf(stringOperator);
+        if (Objects.isNull(operator)) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
정적 팩토리 메서드에서 유효성을 검사하면 어떨까요?

```suggestion
        Operator operator = Operator.symbolOf(stringOperator);
```